hooks/donation/useDonationCheckout.ts

```diff
@@ -122,9 +128,49 @@ export function useDonationCheckout() {
             donations: completedDonations,
             totalProjects: payments.length,
           };
+
           cartState.setLastCompletedSession(session);
-        } else {
-          console.warn("No completed donations to save in session");
+        }
+
+        // Persist successful donations to backend
+        const successfulResults = results.filter((result) => result.status === "success");
+
+        if (successfulResults.length > 0 && address) {
+          Promise.all(
+            successfulResults.map(async (result) => {
+              const payment = payments.find((p) => p.projectId === result.projectId);
+              if (!payment) return;
+
+              const donationRequest: CreateDonationRequest = {
+                uid: `${result.hash}-${payment.projectId}`,
+                chainID: payment.chainId,
+                donorAddress: address,
+                projectUID: payment.projectId,
+                payoutAddress: payoutAddresses[payment.projectId],
+                amount: payment.amount,
+                tokenSymbol: payment.token.symbol,
+                tokenAddress: payment.token.isNative ? undefined : payment.token.address,
+                transactionHash: result.hash,
+                donationType: DonationType.CRYPTO,
+                metadata: {
+                  tokenDecimals: payment.token.decimals,
+                  tokenName: payment.token.name,
+                  chainName: payment.token.chainName,
+                },
+              };
+
+              try {
+                await createDonation(donationRequest);
+              } catch (error) {
+                console.error(
+                  `Failed to persist donation to backend for project ${payment.projectId}:`,
+                  error
+                );
+              }
+            })
+          ).catch((error) => {
+            console.error("Error persisting donations to backend:", error);
+          });
         }
```

Comment:
if this function fails it might still show the user that the donations were executed, and might be better to have a "createBatchDonations" endpoint on the backend rather than looping over donations on the front